multicluster/cmd/check.go

```diff
@@ -579,9 +579,21 @@ func (hc *healthChecker) checkIfGatewayMirrorsHaveEndpoints(ctx context.Context,
 			continue
 		}
 
+		leases, err := hc.KubeAPIClient().CoordinationV1().Leases(multiclusterNs.Name).List(ctx, selector)
```

Comment:
For a non HA mode with only one instance running will it still have a lease resource?

Reply:
Yup. This check should be safe to run regardless of the deployment model. The same applies for `multicluster gateways` where we pull metrics from the "leader". We can guarantee there will always be a leader since a lease is created and claimed under all circumstances.